File: web/src/models/Translation.ts
// Conforms to docs/data/translation.md, spec v1.
import { LanguageCode } from './Common'

/** 'generated' = produced by a translation agent; 'human' = hand-authored/verified. */
export type TranslationSource = 'human' | 'generated'

/**
 * The full-meaning rendering of a single Verse into one human language. A verse may carry
 * translations in several languages; (verse, languageCode) is unique.
 */
export interface ITranslation {
  languageCode: LanguageCode
  /** One or more display lines/paragraphs. Never empty - an absent translation is omitted. */
  text: string[]
  /** Defaults to 'human' when absent from the canonical JSON. */
  source?: TranslationSource
}

File: andorid/app/src/main/java/com/gaudiyakirtan/myapplication/models/Translation.kt
package com.gaudiyakirtan.myapplication.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * The full-meaning rendering of a single [Verse] into one human language.
 * See docs/data/translation.md (spec v1).
 *
 * `source` defaults to "human" when absent, per spec. The shipped corpus never actually sets it
 * (always absent), so every decoded [Translation] currently reports `source == "human"`.
 */
@Serializable
data class Translation(
    @SerialName("language_code") val languageCode: String,
    @SerialName("text") val text: List<String>,
    @SerialName("source") val source: String = "human"
)
